packages/neon_framework/test/dialog_test.dart

```diff
@@ -0,0 +1,171 @@
+  const locale = Locale('en');
+
+  return MaterialApp(
+    theme: theme.lightTheme,
+    localizationsDelegates: NeonLocalizations.localizationsDelegates,
+    supportedLocales: NeonLocalizations.supportedLocales,
+    locale: locale,
+    home: dialog,
+    navigatorObservers: [
+      mockedObserver,
+    ],
+  );
+}
+
+void main() {
+  setUpAll(() {
+    registerFallbackValue(FakeRoute());
+  });
+
+  tearDown(() => reset(mockedObserver));
+
+  group('dialog', () {
+    group('NeonConfirmationDialog', () {
+      testWidgets('NeonConfirmationDialog widget', (final widgetTester) async {
+        const title = 'My Title';
+        var dialog = const NeonConfirmationDialog(title: title);
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        expect(find.text(title), findsOneWidget);
+        expect(find.byType(NeonDialogAction), findsExactly(2));
+
+        // Not true on cupertino platforms
+        expect(find.byType(OutlinedButton), findsOneWidget);
+        expect(find.byType(ElevatedButton), findsOneWidget);
+
+        dialog = const NeonConfirmationDialog(
+          title: title,
+          isDestructive: false,
+        );
+
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        expect(find.byType(NeonDialogAction), findsExactly(2));
+        expect(find.byType(OutlinedButton), findsExactly(2));
+
+        const icon = Icon(Icons.error);
+        const content = SizedBox(key: Key('content'));
+        const confirmAction = SizedBox(key: Key('confirmAction'));
+        const declineAction = SizedBox(key: Key('declineAction'));
+        dialog = const NeonConfirmationDialog(
+          title: title,
+          icon: icon,
+          content: content,
+          confirmAction: confirmAction,
+          declineAction: declineAction,
+        );
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        expect(find.byIcon(Icons.error), findsOneWidget);
+        expect(find.byKey(const Key('content')), findsOneWidget);
+        expect(find.byKey(const Key('confirmAction')), findsOneWidget);
+        expect(find.byKey(const Key('declineAction')), findsOneWidget);
+      });
+
+      testWidgets('NeonConfirmationDialog confirm', (final widgetTester) async {
+        const title = 'My Title';
+        const dialog = NeonConfirmationDialog(title: title);
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        await widgetTester.tap(find.byType(OutlinedButton));
+        verify<void>(() => mockedObserver.didPop(any(), any())).called(1);
+      });
+
+      testWidgets('NeonConfirmationDialog decline', (final widgetTester) async {
+        const title = 'My Title';
+        const dialog = NeonConfirmationDialog(title: title);
+        await widgetTester.pumpWidget(wrapDialog(dialog));
+
+        await widgetTester.tap(find.byType(ElevatedButton));
+        verify<void>(() => mockedObserver.didPop(any(), any())).called(1);
+      });
+    });
```

Comment:
These should test that the expected values are popped and not just any

Reply:
Please show me how.
I already did quite some research and was not able to find a solution to this.
The any only matches the routes and not the attached values.

Reply:
Looking at https://github.com/flutter/flutter/blob/b0b0e423d6ec5f115e0580f9e21ffed9607dd302/packages/flutter/test/material/dialog_test.dart#L363 it seems like you can do this by not pumping the dialog but showing it the normal way using showDialog (and then checking the awaited result).